tests/hw3/task3_test.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+
+from homework3.task3 import mistakes
+
+
+def test_class_filter_positive_case():
+    """
+    Testing that class Filter works correct in positive case
+    """
+    instance_of_filter = mistakes.Filter((lambda a: a % 2 == 0, lambda a: a > 5))
+    assert instance_of_filter.apply(range(10)) == [6, 8]
+
+
+def test_class_filter_negative_case():
+    """
+    Testing that class Filter works correct in negative case
+    """
+    instance_of_filter = mistakes.Filter((lambda a: a % 2 == 0, lambda a: a > 10))
+    assert instance_of_filter.apply(range(10)) == []
+
+
+def test_function_make_filter_positive_case():
+    """
+    Testing that function make_filter works correct in positive case
+    """
+    request = mistakes.make_filter(is_dead=True, kind="parrot").apply(
```

Comment:
Всё-таки в тестах, если есть возможность, лучше использовать данные из требований. Т.е.:
```suggestion
    request = mistakes.make_filter(name='polly', type='bird').apply(
```

конечно чем больше вариантов протестировано, тем лучше, так что в идеале пишутся и свои варианты и те что есть в требованиях.